pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -99,6 +99,9 @@ self: super: builtins.intersectAttrs super {
 
   niv = enableSeparateBinOutput super.niv;
 
+  # Ensure the necessary frameworks for Darwin.
+  OpenAL = addExtraLibrary super.OpenAL (pkgs.lib.optional pkgs.stdenv.isDarwin [ pkgs.darwin.apple_sdk.frameworks.OpenAL ]);
```

Comment:
The difference between `optionals` and `optional` is that the former takes a list argument whereas the latter takes an `element`. Just changing the name of the function won't result in valid code. You have to adapt your arguments, too. You'll notice what I mean when you, like, *test* the patch you're submitting.

Also, if you want to pass a list of dependencies here, then you'll need to use `addExtraLibraries` instead.

Reply:
I'm still able to build the package with the current patch on macos & Linux (`nix-env -f./ -iA haskellPackages.OpenAL`).

Reply:
Yes, you are right. This is an unexpected consequence of the way Nix casts lists into a string by concatenating the individual elements cast to a string. In this particular case, that gives a usable result, i.e. the  `[pkg]` ends up being essentially the same thing as `pkg` (which is what we really want here). It's kind of unfortunate that this bogus code still gives a correct result, mostly due to chance. It's on those occasions where one would really like Nix to have a type system.